src/utp_wrapper.rs

```diff
@@ -56,6 +67,9 @@ impl UtpWrapper {
                                         Err(TryRecvError::Empty) => break,
                                     }
                                 }
+                                if send_keepalive {
+                                    socket.send_keepalive();
```

Comment:
It is outside of the write loop. The only thing inside the write loop is `send_keepalive = false` which tells us not to bother with the keepalive if we've been writing.